src/main/java/com/bloomtech/asylumrgbea/models/CasesQueryParameterDto.java

```diff
@@ -0,0 +1,66 @@
+package com.bloomtech.asylumrgbea.models;
+
+import lombok.Data;
+import org.springframework.lang.Nullable;
+
+import java.util.Objects;
+
+/**
+ * Defines the DTO to store the query parameters provided by the client
+ * for the cases' endpoint. Each attribute represents a query parameter
+ * or category to filter cases by.
+ */
+@Data
+public class CasesQueryParameterDto {
```

Comment:
This Change is much more specific, and I like it.

However, I would encourage this requestDto to be named for what it is requesting, rather than what the request contains. Here are some examples.

TablePageRequest, TablePageRequestDto, PageRequest, QueriedPageRequest, QueriedTablePageRequestDto, or something to that effect.

Reply:
`    /**
     * Defines the GET method pathway for retrieving books
     * @param id String of the book to retrieve
     * @return ResponseEntity<?> get book response
     */
    @GetMapping(value = "/books/{id}", produces = {"application/json"})
    public ResponseEntity<?> getBook(@PathVariable String id) throws Exception {
        GetBookActivity bookActivity = component.provideGetBookActivity();
        GetBookRequest getBookRequest = GetBookRequest.builder().withBookId(id).build();
        return new ResponseEntity<>(bookActivity.execute(getBookRequest), HttpStatus.OK);
    }`
    
    This is the example I was looking through. 
    This

Reply:
The `GET ` Method is used to _request_ data from a specified resource. It contains query parameters.
The `Post` Method is also used to _request_ data, but sends it's request differently. They may or may not be sending request objects, but they are still a request. In comparison a response might be specific to an endpoint, but it is still a response. In previous projects the requests were located in models/requests/ and models/response/ directories making the distinction  more obvious.

The `GET ` Method for `/cases` has specific query parameters in its request. The `GET` Method for a new endpoint lets call it  `/heatmap` may have completely different query parameters, or some shared values. They may be sent as a JSON object, but they are both requests. 

We discussed previously there may be a need in the future for a `QueryFilterParameterModel ` that houses the filter parameters for reuse in other requests to the API. We discussed that it may complicate Spring mapping process to the DTO and decided to leave that design decision to future contributors.

NOTE: In previous projects the requests were located in models/requests/ and models/response/ directories making the distinction between models more obvious.

Reply:
The QueryParameter dto is actually named as CasesQueryParameterDto. It is specific to the cases end point. I would encourage going through the updated sequence diagram or class diagrams if my points are unclear.

I believe you did not address my technical concerns. I understand what Requests are but from the point of OOP, I believed it would be a misnomer to call the components of a Request as the Request itself (ie query parameters, path parameters, and method bodies)

Reply:
I must have read this too quickly, "Request objects are, however, generated using the arguments (query parameters, path parameter, or method bodies) within the Controller Layer. So we could use the `QueryParameterDto` to generate a Request object."

In the case of `GET /cases` it encapsulates everything!

Thank you for the clarification.